zypp/target/RpmPostTransCollector.cc

```diff
@@ -127,7 +127,11 @@ namespace zypp
               npkgs++;
 
             MIL << "EXECUTE posttrans: " << script << " with argument: " << npkgs << endl;
-            ExternalProgram prog( (noRootScriptDir/script).asString() + " " +str::numstring( npkgs ), ExternalProgram::Stderr_To_Stdout, false, -1, true, _root );
+	    ExternalProgram::Arguments cmd;
+	    cmd.push_back( "/bin/sh" );
+	    cmd.push_back( (noRootScriptDir/script).asString() );
+	    cmd.push_back( str::numstring( npkgs ) );
+            ExternalProgram prog( cmd, ExternalProgram::Stderr_To_Stdout, false, -1, true, _root );
```

Comment:
Shouldn't we use initializer lists instead of push_back() if possible?

```suggestion
        ExternalProgram::Arguments cmd {
          "/bin/sh",
          (noRootScriptDir/script).asString(),
          str::numstring( npkgs )    
        };
        ExternalProgram prog( cmd, ExternalProgram::Stderr_To_Stdout, false, -1, true, _root );
```

Reply:
done